src/spec/sync/utils/product.spec.coffee

```diff
@@ -1390,6 +1390,26 @@ describe 'ProductUtils', ->
         expect(actual).toEqual(expected)
 
       it 'should build a transitionState action for a state change', ->
+        # test with { state: null }
+        newRef = _.extend({}, @NEW_REFERENCE, { state: null })
```

Comment:
If i understand it right ```newRef``` will have this structure for product ```{...{"state": null}}``` but it should be ```{...}```. If state is not known than state attribute is not available in new product at all. It should support also state with value null but in most cases state will not be set at all